src/server/set_family.cc

```diff
@@ -52,24 +57,108 @@ intset* IntsetAddSafe(string_view val, intset* is, bool* success, bool* added) {
   return is;
 }
 
-unsigned AddStrSet(ArgSlice vals, CompactObj* dest) {
-  unsigned res = 0;
+bool dictContains(const dict* d, string_view key) {
+  uint64_t h = dictGenHashFunction(key.data(), key.size());
+
+  for (unsigned table = 0; table <= 1; table++) {
+    uint64_t idx = h & DICTHT_SIZE_MASK(d->ht_size_exp[table]);
+    dictEntry* he = d->ht_table[table][idx];
+    while (he) {
+      sds dkey = (sds)he->key;
+      if (sdslen(dkey) == key.size() &&
+          (key.empty() || memcmp(dkey, key.data(), key.size()) == 0)) {
+        return true;
+      }
+      he = he->next;
+    }
+    if (!dictIsRehashing(d)) {
+      break;
+    }
+  }
+  return false;
+}
 
-  dict* ds = (dict*)dest->RObjPtr();
-  auto* es = EngineShard::tlocal();
+pair<unsigned, bool> RemoveStrSet(ArgSlice vals, CompactObj* set) {
+  unsigned removed = 0;
+  bool isempty = false;
+  bool use_set2 = GetFlag(FLAGS_use_set2);
```

Comment:
i prefer that we read a flag in a single place - InitStrSet and in all other places pls test the encoding.